Updating Compute Diagnostic RP's swagger with additional APIs

ansahdev opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [x] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

ansahdev committed:
Updating Compute Diagnostic RP's swagger with additional APIs

ansahdev committed:
Updating Compute Diagnostic RP's swagger with additional APIs

tjegbejimba committed:
Add missing PrivateLink Operations (#18030)

* fix privateLink client name

* Add missing PL operations

* fix prettier and validation errors

* wrap PrivateLinkAssociation in properties field

* match pla put and get response

* fix getpla example

* add 201 response to PUT requests

* add 201 response to PUT examples

* Update specification/resources/resource-manager/Microsoft.Authorization/stable/2020-05-01/privateLinks.json

Co-authored-by: Dapeng Zhang <ufo54153@gmail.com>

Co-authored-by: Dapeng Zhang <ufo54153@gmail.com>

darkoa-msft committed:
[Microsoft.DeviceUpdate] (Internal only) Added Private Endpoint Proxy Private Endpoint Update (#18281)

* (Internal only) Added Private Endpoint Proxy Private Endpoint Update

* The existing resource also needs a change

* Renamed example

* Fixing more copy and paste errors

* Removed read only, formatted new file

* Move provisioning state inside proxy properties

* Revert "Move provisioning state inside proxy properties"

This reverts commit 414d0c687ca3420426606d0f4d54923ee58ddb82.

* Moved proxy provisioning state into properties, try #2

* Moved proxy provisioning state into properties, try number 3

* Moved proxy provisioning state into properties, try number 4

* Moved Proxy provisioningState to top level, removed property bag

* Fixed a naming error

* Linter fixes

* Removed flatten in Proxy properties

* Reverted all provisioningState changes

* Swapped properties to match the old swagger

ArcturusZhang committed:
add x-ms-enum to trafficmanager (#17511)

LinglingTong committed:
add xms-ids for PowerBIdedicated (#17971)

LinglingTong committed:
add xms-ids for portal (#17968)

LoadBearingBean committed:
Add x-ms-identifier to relay\resource-manager (#17760)

Co-authored-by: Steven Vukelich <svukel@microsoft.com>

gaoyp830 committed:
[Hub Generated] Review request for Microsoft.Capacity to add version stable/2022-03-01 (#18244)

* Adds base for updating Microsoft.Capacity from version stable/2021-07-01 to version 2022-03-01

* Updates readme

* Updates API version in new specs and examples

* Update with new changes

* include quota.json in readme

* Remove dollar sign for publisher id

* Change name from FulfillmentCreatedDateTime to BenefitStartTime

FumingZhang committed:
Review request for Microsoft.ContainerService to add version 2022-02-02-preview (#18150)

* Adds base for updating Microsoft.ContainerService from version preview/2022-01-02-preview to version 2022-02-02-preview

* Updates readme

* Updates API version in new specs and examples

* Add key management service profile of a managed cluster for version 2022-02-02-preview (#17955)

* Add key management service profile of a managed cluster for version 2022-02-02-preview

* add kmsProfile

* update after api review

* add more details in description

* update for comments

* add mc snapshot api (#17974)

* add mc snapshot api

* fix format

* add mc snapshot type

* define type for enums

* add properties for example

* Update readmes for the 2022-02-02-preview dev branch of container service (#17889)

* update readme

* remove useless -only tags

* remove nested property in allOf for 2022-02-02-preview managedClusters (#18063)

* Fix credential format typo in 0102 preview (#18117)

* add missing type object (#18116)

* Update several common definitions according to arm feedback for container service 2022-02-02-preview (#18206)

* update definition of serveral parameters and system data

* update definition of Resource and move property location to sub resources

* add missing tags property for prev Resource

* fix for compatibility

* fix definition name

Co-authored-by: Bin Xia <binxi@microsoft.com>
Co-authored-by: Qingqing <qizhe@microsoft.com>
Co-authored-by: Tongyao Si <tosi@microsoft.com>

LinglingTong committed:
Add xms-ids for monitor (#17851)

* add xms-ids for monitor

* inner properties

* change operator to name

satravi committed:
Added new API version 2022-03-01 to the Mobile Network service (#18234)

* Mobile Network REST API Specification

* Removed API versions 2020-06-01-preview and 2021-04-01-preview. Have only the API version 2022-01-01-preview in Public

* Addressed Swagger Avocado and Swagger SpellCheck Errors

* Addressed Parameters Order Errors

* Merge azure-rest-api-specs-pr branch ssivathas/mobilenetwork-public-api into satravi/mobilenetwork-rest-api-specs

* Removed Unreferenced JSON Files

* Added support for Static IP Addressing to PMN

* Addressed Swagger PrettierCheck Errors

* Addressed Swagger ModelValidation Errors

* change go config to track 2

* Updated the API versions in SDK README files

* Update readme.python.md

* Update readme.md

* Updated ActivationState to ConfigurationState

* Addressed XmsIdentifierValidation Errors

* Addressed DescriptionAndTitleMissing Errors

* Generic name for N2 and N3 Interface attributes to support 4G Networks

* Generic name for N6 Interface attribute to support 4G Networks

* Generic name for N6 Interface attribute to support 4G Networks

* Updated the Mobile Network REST specification and examples

* Removed the read-only field activationState

* Added 2022-03-01 version

* Updated 2022-03-01 version in the README files

* Removed the duplicate custom-words

* Revert "Mobile Network API version 2022-03-01"

* Revert custom-words

* Added Mobile Network API version 2022-03-01-preview

* Updated Mobile Network API version 2022-03-01-preview specs and examples

Co-authored-by: ArcturusZhang <dapzhang@microsoft.com>
Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>

SapirElmakayes committed:
[Hub Generated] Public private branch 'dev-security-Microsoft.Security-2021-12-01-preview' (#18267)

* Adds base for updating Microsoft.Security from version preview/2021-07-01-preview to version 2021-12-01-preview

* Updates readme

* Updates API version in new specs and examples

* Keep only Securityconnectors in the API version

* Adding CspmMonitorGcp offering

* minors

* Adding Security connector

* Fix prettier issue

* Adding information protection offering

* Adding suppression rule

* added github configuration, fixed bugs

* updated examples

* added example

* fixed bugs in offering inheritnece, added org data for aws/gcp objects, updated discriminators

* added discriminator for orgmembershipType on environmentData object, updated aws to awsAccount on examples for aws env data

* fixed discriminator position

* all example validations + schema validations pass, BEFORE adding updated examples

* fixed null aws env data property

* Adding DefenderForDatabases

* Run prettier

* Small renamings

* Fix native cloud conection

* update servers offerings

* adding containers

* Remove DefenderForDatabases offerings

* Adding Qualys as known word

* Update the VA type to enum

* small changes for servers offering and gcpproject properties

* added missing enum types

* Servers P1P2

* Add P1P2 to servers GCP

* subPlan fixes

* Remove supression

* add suppression rule back

* Adding operations to the default tag

* Add x-ms-identifiers

Co-authored-by: Sapir Elmakayes <saelmaka@microsoft.com>
Co-authored-by: chemishumacher <chemishumacher@gmail.com>

maorleger committed:
[Hub Generated] Review request for Microsoft.KeyVault to add version stable/7.3 (#18028)

* Adds base for updating Microsoft.KeyVault from version preview/7.3-preview to version 7.3

* Updates readme

* Updates API version in new specs and examples

* [KeyVault] - Add exclusion for known spellcheck issue (#18031)

* add exclusion for known spellcheck issue

* collapse cspell ignore for keyvault's storage.json

giakas committed:
Dev mediaservices microsoft.media 2021 11 01 (#17540)

* Adds base for updating Microsoft.Media from version stable/2021-06-01 to version 2021-11-01

* Updates readme

* Updates API version in new specs and examples

* Update Account swagger to 2021-11-01

* add the supressions

* remove descriminator for Layers

* use 2021-06-01 version for encoding service

* remove 404 response

* Revert "use 2021-06-01 version for encoding service"

This reverts commit f1e9a604b429e96b5be4bdeecfcf8875a47ef531.

* 202 for patch suprression

* Update privateLinks to v3 to get systemData

* Revert "Update privateLinks to v3 to get systemData"

This reverts commit c43f85720c4328057197afd64c8efd2c153670df.

* update streaming service and assetTracks swagger

* fix lint errors.

* fix some lint check.

* rename examples

* removing patch supression test

* Update asset-tracks-operation-result-by-id-no-content.json

* Update AssetsAndAssetFilters.json

set x-ms-paging with null in list tracks

* Update asset-tracks-operation-result-by-id-no-content.json

remove 202 in post example

* Update asset-tracks-operation-result-by-id.json

add 202 back

* Revert "removing patch supression test"

This reverts commit f8979223d791e06ffa2c846add52a6b28affca2a.

* remove operation result data type

* update description on a suppression.

* add another path to the systemData suppression

* update example and description

* Update AssetsAndAssetFilters.json

update api description

* Add async headers to examples

* add async headers to live event and streaming endpoint examples

* Splitting Account service changes from this PR

* remove unused account examples

* Rebase with x-ms-identifier changes.

* fix a lint error for x-ms-identifier

Co-authored-by: zimao <zimao@microsoft.com>
Co-authored-by: Ziwei Mao <19548067+ZzzMao@users.noreply.github.com>

pilor commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -312,6 +404,33 @@
       },
       "description": "Api output result when disk inspection result is completed."
     },
+    "StorageConfigurationInput": {
```
what linkedAccessChecks are performed on the properties in this request?

pilor commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -206,6 +206,98 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection/validateStorageConfiguration": {
```
"registering" a storage config appears to only register it for a single resource (the "resourceId" property). What is "validate" actually validating?  That ANY resource in the subscription and location has a storage config?

ansahdev replied:
After reviewing the existing functionality, we have agreed to not have "resourceId" information passed in as part of "registerStorageConfiguration" API. The "registerStorageConfiguration" API is intended to create a configuration entry between **subscriptionId -> storageAccount (SubscriptionLevel)**. "ValidateStorageConfiguration" will lookup if such a configuration exists for a subscription and perfom RBAC checks on SA for caller before sending response. I will update swagger and send out another iteration with the change.

pilor commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -206,6 +206,98 @@
       }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection/registerStorageConfiguration": {
+      "post": {
```
using this POST model means that users will not be able to set up their subscription completely using just a template deployment. That means they will have to perform imperative calls via portal, powershell, cli, or similar which is something that should typically be avoided for configuration like this.

pilor commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -312,6 +404,33 @@
+          "description": "Qualified name of the resource.",
+          "type": "string"
+        },
+        "storageAccountId": {
```
what is the storage account actually used for and how is it accessed?  What identity is used to get the keys to the storage account if that is the mechanism used to access it? What about storage accounts that have disabled key based auth?

ansahdev replied:
1) The new 2 APIs (RegisterStorageConfiguration and ValidateStorageConfiguration) augment the "run" action which executes disk Inspection. These enable a capability where customer need not "remember" which storage account to store results in. StorageAccount information is pinned at subscription level. Details are as follows:
2) The "run" action takes in an "uploadSasUri" parameter as an input in request body. After a successful execution, the results are uploaded to the blob container which is pointed by the input uploadSasUri. This input value is generated outside of RP -> by the caller, either programmatically in Azure Portal using logged in user's identity and accesses or through CLI/PS/SDK where the value is passed in. It can be generated using system key or user-based key as long as it has write permissions. 
2) All the actions call in to /CheckAccess API upfront to validate if caller has correct set of permissions to invoke the underneath use case on the resourceId -> have read permissions on the resourceId to fetch logs from, have read/write permisisons on Storage account to work with.
3) The new APIs (registerStorageConfiguration and ValidateStorageConfiguration) are designed to remember the last known used storage account, similar to cached value so caller need not remember where to have the results uploaded. This will help in E2E Azure Portal experience, where customer need not "remember" or always "select" a storage account from the dropdown control. ValidteStorageConfiguration will return storageaccount id associated at the subscription level, if logged in caller has access to write to it (leverage /CheckAccess API). If not, an error indicating the same is returned through the API. Different calling clients will handle the error differently, either navigate to "Update settings" section or customer call invoke the API through cli by providing a sas that they have access to.

pilor replied:
gotcha.  What is the "resourceId" in the setConfig action for if the storage account is configured for the entire subscription?

pilor replied:
Also, what is the "location" URI segment for in registerStorageConfig?  1 config per location per subscription is possible? When validating you choose the same location as the entity that diag is being run against?

ansahdev replied:
1) After reviewing the existing functionality, we have agreed to not have "resourceId" information passed in as part of "registerStorageConfiguration" API. The "registerStorageConfiguration" API is intended to create a configuration entry between subscriptionId -> storageAccount (SubscriptionLevel). "ValidateStorageConfiguration" will lookup if such a configuration exists for a subscription and perfom RBAC checks on SA for caller before sending response. I will update swagger and send out another iteration with the change
2) Compute Diagnostic RP is a proxy based, location based routing RP. Hence the path has /location/ for all actions. For Private preview, register action is for 1 config per subscription. Post Public preview, we intend to work off customer feedback to add 1 config per location per subscription changes to have regional affinity of data.

darkoa-msft committed:
[Microsoft.DeviceUpdate][Internal Only] Fixed RPaaS linter error - provisioningState (#18328)

* Fixed RPaaS linter error - provisioningState

* Linter fixes

* Added description

sharadkumr committed:
Skumar/resource health r2018 linting (#18310)

* Defining the APIs for the new NSP resource

* fixed R2018 linting error in ResourceHealth.json

Co-authored-by: Hari Prasad Perabattula <harics24@users.noreply.github.com>
Co-authored-by: Hari Prasad Perabattula <haperaba@microsoft.com>

BigCat20196 committed:
update readme python (#18282)

sasank-chaganty committed:
making primary domain controller optional (#18278)

angouloure committed:
updating the DnsNameLabelReusePolicy to what users should see (#18081)

mveraluna committed:
Dropping not from budget filter. (#18309)

Co-authored-by: Maximiliano Vera Luna <mveraluna@microsoft.com>

LinglingTong committed:
add xms-ids for operationsManagement (#17950)

kalyanrk61 committed:
synapse web activity update: Disable Cert Validation property. (#18294)

* Add pipeline policy and ElapsedTime Policy

* adding missing objects

* fix schema validation

* fix prettier issues

* add disable cert validation in webactivity

Co-authored-by: Kalyan C Kesireddy <kalyanke@microsoft.com>

FumingZhang committed:
Review request for Microsoft.ContainerService to add version 2022-02-01 (#18324)

* Adds base for updating Microsoft.ContainerService from version stable/2022-01-01 to version 2022-02-01

* Updates readme

* Updates API version in new specs and examples

* Update readmes for the 2022-02-01 dev branch of container service (#17887)

* update readme

* remove useless -only tags

* Fix violated rule R2026 for 2022-02-01 managedCluster swagger (#18024)

* fix allof

* add missing type

* GA alias minor version (#18038)

* GA alias minor version

* polish more details on kubernetesVersion and currentKubernetesVersion

* Update specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-02-01/managedClusters.json

minor revision

Co-authored-by: Matthew Christopher <matthchr@microsoft.com>

Co-authored-by: Matthew Christopher <matthchr@microsoft.com>

* add missing type object (#18115)

* Revert "GA alias minor version (#18038)" (#18291)

This reverts commit 3a99cda3cb48e05c74923f5467c0737014322b0f.

* Align modifications of several common definitions with 2022-02-02-preview for container service 2022-02-01 (#18216)

* sync modification

* fix ref

Co-authored-by: Jianping Zeng <zjpjack@users.noreply.github.com>
Co-authored-by: Matthew Christopher <matthchr@microsoft.com>

nitsi committed:
New API version to Microsoft.Security 2021-11-01 (#18045)

* Adds base for updating Microsoft.Security from version stable/2021-01-01 to version 2021-11-01

* Updates readme

* Updates API version in new specs and examples

* New API version to Microsoft.Security

pick 5459ef18a7 New API version to Microsoft.Security
pick c8462f9c60 [Microsoft.Security alerts] fix examples/Alerts/SimulateAlerts_example.json
pick b5f4550d6b [Microsoft.Security alerts] add missing API to readme.md

* [Microsoft.Security alerts] fix examples/Alerts/SimulateAlerts_example.json

* [Microsoft.Security alerts] add missing API to readme.md

* Fix readme.md file after rebase

* Removing "x-ms-long-running-operation" header

* Setting target package back to package-composite-v3

* Updating securityContacts.json to the latest version

* Reverting securityContacts.json to equal main, should be updated by the relevant team

Co-authored-by: Nitsan Bracha <nibracha@microsoft.com>

colawwj committed:
add-typescipt.md-for-hardwaresecuritymodules (#18333)

ArchangelSDY committed:
Add Microsoft.SignalRService/signalr 2022-02-01 (#17748)

* Prepare Microsoft.SignalRService/signalr 2022-02-01

* Add Microsoft.SignalRService/signalr 2022-02-01

* Address review comments

* Fix CI

ansahdev committed:
Merge branch 'feature/cdrp-2021-06-01-preview-newAPIs' of https://github.com/Azure/azure-rest-api-specs into feature/cdrp-2021-06-01-preview-newAPIs

ansahdev committed:
1) Prettier check fix
2) Udpate StorageConfigurationInout data model -> removed "resourceId" property.

ansahdev commented:
Closing this Pull request and creating a new one.
